### PR TITLE
log: Ensure threaded eve honors SIGHUP

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #5382 

This commit ensures that all logging contexts register for the file
rotation mechanism (SIGHUP and configured) and that the suffixes use for threaded EVE output are continuous.

Describe changes:
- Ensure EVE suffixes are continuuous.

Redmine issue:
https://redmine.openinfosecfoundation.org/issues/3915

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

File: src/util-logopenfile.c [hunk 1]
```diff
@@ -713,15 +716,22 @@ LogFileCtx *LogFileEnsureExists(LogFileCtx *parent_ctx, int thread_id)
  */
 static bool LogFileNewThreadedCtx(LogFileCtx *parent_ctx, const char *log_path, const char *append, int thread_id)
 {
+    static bool eve_suffix_init = false;
```
Comment (victorjulien) [inline src/util-logopenfile.c:719]: is this shared between threads as well? If so it should probably be an atomic too?

**Comment by jasonish** [comment]
First thing I noticed is that this starts with eve.json.0, which is fine on its own.  However, our pcap.log starts at 1. I wonder if we should be consistent here.

File: src/util-logopenfile.c (hunk 1 above)
Comment (jasonish) [reply src/util-logopenfile.c:719]: Looks like it is.

**Review by jasonish — COMMENTED** [review]
Tests good, but it looks like that one var should be atomic.

@victorjulien Any thoughts on the thread id numbering? I think I'd like to be consistent with the pcap-log and start at 1. But OK to leave it at 0 and never think of it again.

**Comment by victorjulien** [comment]
> @victorjulien Any thoughts on the thread id numbering? I think I'd like to be consistent with the pcap-log and start at 1. But OK to leave it at 0 and never think of it again.

I prefer it to be consistent, so lets start at 1.

**Comment by jlucovsky** [comment]
Continued in #5399

**Pull request closed without merging**